R/zchunk_LB134.Diet_Rfao.R

```diff
@@ -0,0 +1,384 @@
+      mutate(value = value * kcalkg / conv_d) %>%
+      select(FAO2050_reg, FAO2050_item, GCAM_demand, year, value) ->
+      L134.Diet_kcald_Rfao_Cfao_Y
+
+    # Aggregate by GCAM demand and FAO region, and compute future diet ratios by FAO2050 region
+    # Original lines 94-104
+    L134.Diet_kcald_Rfao_Cfao_Y %>%
+      filter(year %in% aglu.DIET_YEARS, !is.na(GCAM_demand)) %>%
+      group_by(FAO2050_reg, GCAM_demand, year) %>%
+      summarise(value = sum(value)) %>%
+      ungroup %>%
+      mutate(GCAM_demand = if_else(GCAM_demand == "total", "crops", GCAM_demand)) ->
+      L134.Diet_kcald_Rfao_Dmnd_Y
+
+    # Note on this next section from Page:
+    # Can't just use exogenous caloric contents for the crop and meat commodities, which we're estimating
+    # as averages without knowing the exact composition of the commodities (e.g., what portion of "cereals"
+    # is corn vs wheat in each region, as they have different caloric contents), and still hit the target
+    # reported "Total food (kcal/person/day)" in each region and time period. There were three options:
+    # (1) assign the crop caloric contents exogenously and allow the meat caloric contents to float;
+    # (2) vice versa; or
+    # (3) assign both crop and meat caloric contents, and then scale them all to hit the target value.
+    # It looks like I went with (2), maybe because I figured that the meat commodities probably have less
+    # heterogeneity in average caloric contents, and maybe also because it happened to be the easiest to implement.
+
+    # Compute diet ratios based on meat caloric contents and demand (option 2 above)
+    L134.Diet_kcald_Rfao_Dmnd_Y %>%
+      filter(GCAM_demand == "meat") %>%
+      select(-GCAM_demand) %>%
+      rename(meat_value = value) %>%
+      right_join(L134.Diet_kcald_Rfao_Dmnd_Y, by = c("FAO2050_reg", "year")) %>%
+      mutate(value = if_else(GCAM_demand == "crops", value - meat_value, value)) %>%
+      select(-meat_value) %>%
+      # compute future diet ratios by FAO2050 region
+      group_by(FAO2050_reg, GCAM_demand) %>%
+      arrange(year) %>%
+      mutate(value = value / first(value)) ->
+      L134.DietRatio_Rfao_Dmnd_Y
+
+    # Multiply these ratios by the starting values (final historical year value, fhy_value) at the country level
+    # Original lines 106-110
+    L134.DietRatio_Rfao_Dmnd_Y %>%
+      left_join(rename(L134.Food_t_ctry_Dmnd_Y, fhy_value = value), by = c("FAO2050_reg", "GCAM_demand")) %>%
+      mutate(value = value * fhy_value) %>%
+      select(-fhy_value) ->
+      L134.Food_t_ctry_Dmnd_Y
+
+    # Match in GCAM regions, aggregate, and compute ratios from final historical year
+    # Original lines 112-125
+    L134.Food_t_ctry_Dmnd_Y %>%
+      left_join_error_no_match(select(iso_GCAM_regID, iso, GCAM_region_ID), by = "iso") %>%
+      group_by(GCAM_region_ID, GCAM_demand, year) %>%
+      summarise(value = sum(value)) ->
+      L134.Food_t_R_Dmnd_Y
+
+    L134.Food_t_R_Dmnd_Y %>%
```

Comment:
Is there a reason why this is a separate pipeline? The comment explaining it is above the previous.